### PR TITLE
chore(release): prepare for v2.7.1

### PR DESCRIPTION
**Pull request by JacobCoffee**
## Description

- release 2.7.1

### CODE REVIEW

File: docs/release-notes/changelog.rst [hunk 1]
```diff
@@ -3,6 +3,102 @@
...
+
+        This addresses an issue when serializing `Enums` that was reported in discord.
+
+    .. change:: replace TestClient.__enter__ return type with Self
+        :type: bugfix
+        :pr: 3194
+
+        ``TestClient.__enter__`` and ``AsyncTestClient.__enter__`` return ``self``.
+        If you inherit ``TestClient``, its ``__enter__`` method should return derived class's instance
+        unless override the method. So I think :class:`~typing_extensions.Self` could be more flexible return type.
```
Comment (Alc-Alc) [inline docs/release-notes/changelog.rst:21]: > So I think

should this part be removed / reworded?

File: docs/release-notes/changelog.rst [hunk 2]
```diff
@@ -3,6 +3,102 @@
...
+.. changelog:: 2.7.1
+    :date: 2024-03-22
+
+    .. change:: add default encoders for `Enums` and `EnumMeta`
+        :type: bugfix
+        :pr: 3193
+
+        This addresses an issue when serializing `Enums` that was reported in discord.
+
+    .. change:: replace TestClient.__enter__ return type with Self
```
Comment (Alc-Alc) [inline docs/release-notes/changelog.rst:15]: should we use double backticks here? I see some more single backticks, ignore these if I am wrong

File: docs/release-notes/changelog.rst [hunk 3]
```diff
@@ -3,6 +3,102 @@
...
+        Python 3.12 introduced a new way to configure ``QueueHandler`` and ``QueueListener`` via
+        :meth:`~logging.config.dictConfig()`. As described in the
+        `logging documentation <https://docs.python.org/3/library/logging.config.html#configuring-queuehandler-and-queuelistener>`_.
+
+        The listener still needs to be started & stopped, as previously.
+        To do so, we've introduced ``LoggingQueueListener``.
+
+        And as stated in the doc:
+        * Any custom queue handler and listener classes will need to be defined with the same initialization signatures
+        as `QueueHandler <https://docs.python.org/3/library/logging.handlers.html#logging.handlers.QueueHandler>`_ and
```
Comment (Alc-Alc) [inline docs/release-notes/changelog.rst:98]: same

File: docs/release-notes/changelog.rst [hunk 4]
```diff
@@ -3,6 +3,102 @@
...
+        :type: bugfix
+        :pr: 3196
+        :issue: 3047
+
+        This specifies the ``spec-url`` and ``apiDescriptionUrl`` of Rapidoc, and Stoplight Elements as absolute
+        paths relative to the root of the site.
+
+        This ensures that both of the send the request for the JSON of the OpenAPI schema to the right endpoint.
+
+    .. change:: JSON schema `examples` were OpenAPI formatted
```
Comment (Alc-Alc) [inline docs/release-notes/changelog.rst:33]: same

**Review by Alc-Alc — APPROVED** [review]

File: docs/release-notes/changelog.rst [hunk 5]
```diff
@@ -3,6 +3,102 @@
...
 =============
 
+.. changelog:: 2.7.1
+    :date: 2024-03-22
+
+    .. change:: add default encoders for `Enums` and `EnumMeta`
+        :type: bugfix
+        :pr: 3193
+
+        This addresses an issue when serializing `Enums` that was reported in discord.
```
Comment (JacobCoffee) [inline docs/release-notes/changelog.rst:13]:
```suggestion
        This addresses an issue when serializing ``Enums`` that was reported in discord.
```

File: docs/release-notes/changelog.rst (hunk 1 above)
Comment (JacobCoffee) [inline docs/release-notes/changelog.rst:21]:
```suggestion
        unless override the method. :class:`~typing_extensions.Self`is a more flexible return type.
```

File: docs/release-notes/changelog.rst (hunk 4 above)
Comment (JacobCoffee) [inline docs/release-notes/changelog.rst:33]:
```suggestion
    .. change:: JSON schema ``examples`` were OpenAPI formatted
```

File: docs/release-notes/changelog.rst (hunk 3 above)
Comment (JacobCoffee) [reply docs/release-notes/changelog.rst:98]: Not sure what you mean on this last one

**Review by JacobCoffee — COMMENTED** [review]
resolved all but one comment

File: docs/release-notes/changelog.rst (hunk 3 above)
Comment (Alc-Alc) [reply docs/release-notes/changelog.rst:98]: I think I made a mistake, no backticks for sphinx stuff?
```
`~logging.config.dictConfig()`
```
?

Comment (JacobCoffee) [reply docs/release-notes/changelog.rst:98]: only one when using roles or links like this if thats wym?

**Comment by JacobCoffee** [comment]
Theres quite a few new things added, ill need to re-run the release process and combine

**Pull request merged**